pkgs/auto-fix-vscode-server.nix

```diff
@@ -1,94 +1,152 @@
+  nodejsFHS = buildFHSUserEnv ({
+    name = "node";
+    targetPkgs = _: runtimeDependencies;
+    extraBuildCommands = ''
+      if [[ -d /usr/lib/wsl ]]; then
+        cp -rsHf /usr/lib/wsl usr/lib/wsl
+      fi
+    '';
+    runScript = "${nodejs}/bin/node";
     meta = {
       description = ''
         Wrapped variant of Node.js which launches in an FHS compatible envrionment,
-        which should allow for easy usage of extensions without nix-specific modifications.
+        which should allow for easy usage of extensions without Nix-specific modifications.
       '';
     };
-  };
+  });
 
-  nodejsWrapped = if enableFHS then nodejsFHS else nodejs;
+  patchELFScript = writeShellScript "patchelf-vscode-server.sh" ''
+    set -euo pipefail
+    PATH=${makeBinPath [ coreutils findutils patchelf ]}
+    INTERP=$(< ${stdenv.cc}/nix-support/dynamic-linker)
+    RPATH=${makeLibraryPath runtimeDependencies}
+    bin_dir=$1
 
-in writeShellScript "auto-fix-vscode-server.sh" ''
-  set -euo pipefail
-  PATH=${lib.makeBinPath [ coreutils findutils inotify-tools patchelf ]}
+    # NOTE: We don't log here because it won't show up in the output of the user service.
 
-  bins_dir=${installPath}/bin
-  node_interp=$(patchelf --print-interpreter ${nodejs}/bin/node)
-  node_rpath=$(patchelf --print-rpath ${nodejs}/bin/node)
+    # Check if the installation is already full patched.
+    if (( $(< "$bin_dir/.patched") )); then
+      continue;
+    fi
 
-  patch_bin() {
-    local bin_dir=$1 interp
-    ln -sfT ${nodejsWrapped}/bin/node "$bin_dir/node"
-    while read -rd ''' bin; do
+    while read -rd ''' elf; do
       # Check if binary is patchable, e.g. not a statically-linked or non-ELF binary.
-      if ! interp=$(patchelf --print-interpreter "$bin" 2>/dev/null); then
+      if ! interp=$(patchelf --print-interpreter "$elf" 2>/dev/null); then
         continue
       fi
+
       # Check if it is not already patched for Nix.
-      if [[ $interp == "$node_interp" ]]; then
+      if [[ $interp == "$INTERP" ]]; then
         continue
       fi
-      patchelf --set-interpreter "$node_interp" --set-rpath "$node_rpath" "$bin"
-      patchelf --shrink-rpath "$bin"
+
+      # Patch the binary based on the binary of Node.js,
+      # which should include all dependencies they might need.
+      patchelf --set-interpreter "$INTERP" --set-rpath "$RPATH" "$elf"
+
+      # The actual dependencies are probably less than that of Node.js,
+      # so shrink the RPATH to only keep those that are actually needed.
+      patchelf --shrink-rpath "$elf"
     done < <(find "$bin_dir" -type f -perm -100 -printf '%p\0')
+
+    # Mark the bin directory as being fully patched.
+    echo 1 > "$bin_dir/.patched"
+  '';
+
+in writeShellScript "auto-fix-vscode-server.sh" ''
+  set -euo pipefail
+  PATH=${makeBinPath [ coreutils findutils inotify-tools ]}
+  bins_dir=${installPath}/bin
+
+  patch_bin () {
+    local bin=$1 actual_dir=$bins_dir/$1 bin_dir
```

Comment:
the value of actual-dir is nonsense:
`/home/<SNIP>/.vscode-server/bin//home/<SNIP>/.vscode-server/bin/441438abd1ac652551dbe4d408dfcec8a499b8bf-1676396781`

This is probably specific to the inotifywait part.

Reply:
I thought I had fixed both call sites to only pass along `$bin`, fixed.